tests/nativeapp/test_post_deploy_for_app.py

```diff
@@ -172,53 +168,20 @@ def test_sql_scripts_with_no_warehouse_no_database(
 @mock.patch(SQL_EXECUTOR_EXECUTE)
 @mock_connection()
 def test_missing_sql_script(
-    mock_execute_query,
-    mock_conn,
-    project_directory,
+    mock_execute_query, mock_conn, project_directory, workspace_context
 ):
     mock_conn.return_value = MockConnectionCtx()
     with project_directory("napp_post_deploy_missing_file_v2") as project_dir:
         dm = DefinitionManager()
         app_model: ApplicationEntityModel = dm.project_definition.entities["myapp"]
+        app = ApplicationEntity(app_model, workspace_context)
 
         with pytest.raises(MissingScriptError) as err:
-            ApplicationEntity.execute_post_deploy_hooks(
-                console=cc,
-                project_root=dm.project_root,
-                post_deploy_hooks=app_model.meta.post_deploy,
-                app_name=app_model.fqn.name,
-                app_warehouse=app_model.meta.warehouse or "MockWarehouse",
-            )
+            app.execute_post_deploy_hooks()
 
         assert err.value.message == 'Script "scripts/missing.sql" does not exist'
 
 
-@mock.patch(SQL_EXECUTOR_EXECUTE)
-@mock_connection()
-def test_invalid_hook_type(
```

Comment:
it was previously possible to call `ApplicationEntity.execute_post_deploy_hooks` with an invalid hook type, but now that it's an instance method, it requires a valid `ApplicationEntityModel` instance, which doesn't allow loading or setting an invalid hook type, so this test is no longer required (or possible)

Reply:
awesome!